src/uu/shuf/src/shuf.rs

```diff
@@ -10,132 +10,165 @@
 #[macro_use]
 extern crate uucore;
 
+use clap::{App, Arg};
 use rand::Rng;
 use std::fs::File;
 use std::io::{stdin, stdout, BufReader, BufWriter, Read, Write};
-use std::usize::MAX as MAX_USIZE;
+
+struct Options {
+    head_count: usize,
+    output: Option<String>,
+    random_source: Option<String>,
+    repeat: bool,
+    sep: u8,
+}
 
 enum Mode {
-    Default,
-    Echo,
+    Default(String),
+    Echo(Vec<String>),
     InputRange((usize, usize)),
 }
 
 static NAME: &str = "shuf";
 static VERSION: &str = env!("CARGO_PKG_VERSION");
+static OPT_ECHO: &str = "echo";
+static OPT_INPUT_RANGE: &str = "input-range";
+static OPT_HEAD_COUNT: &str = "head-count";
+static OPT_OUTPUT: &str = "output";
+static OPT_RANDOM_SOURCE: &str = "random-source";
+static OPT_REPEAT: &str = "repeat";
+static OPT_ZERO_TERMINATED: &str = "zero-terminated";
+static OPT_FILE: &str = "file";
+static USAGE: &str = r#"shuf [OPTION]... [FILE]
+  or:  shuf -e [OPTION]... [ARG]...
+  or:  shuf -i LO-HI [OPTION]...
+Write a random permutation of the input lines to standard output.
 
-pub fn uumain(args: impl uucore::Args) -> i32 {
-    let args = args.collect_str();
+With no FILE, or when FILE is -, read standard input.
+"#;
 
-    let mut opts = getopts::Options::new();
-    opts.optflag("e", "echo", "treat each ARG as an input line");
-    opts.optopt(
-        "i",
-        "input-range",
-        "treat each number LO through HI as an input line",
-        "LO-HI",
-    );
-    opts.optopt("n", "head-count", "output at most COUNT lines", "COUNT");
-    opts.optopt(
-        "o",
-        "output",
-        "write result to FILE instead of standard output",
-        "FILE",
-    );
-    opts.optopt("", "random-source", "get random bytes from FILE", "FILE");
-    opts.optflag("r", "repeat", "output lines can be repeated");
-    opts.optflag("z", "zero-terminated", "end lines with 0 byte, not newline");
-    opts.optflag("h", "help", "display this help and exit");
-    opts.optflag("V", "version", "output version information and exit");
-    let mut matches = match opts.parse(&args[1..]) {
-        Ok(m) => m,
-        Err(f) => crash!(1, "{}", f),
-    };
-    if matches.opt_present("help") {
-        let msg = format!(
-            "{0} {1}
+pub fn uumain(args: impl uucore::Args) -> i32 {
+    let matches = App::new(executable!())
+        .name(NAME)
+        .version(VERSION)
+        .template("Usage: {usage}\nMandatory arguments to long options are mandatory for short options too.\n{unified}")
+        .usage(USAGE)
+        .help_short("")
```

Comment:
Setting this as empty will override the short auto-generated help.

Reply:
Ah, i see now there is no `-h` for shuf :)

Reply:
Oops, this didn't actually work and I forgot to remove it. I'll remove it. Thanks!